shared/src/akri/configuration.rs

```diff
@@ -28,6 +29,21 @@ pub struct DiscoveryHandlerInfo {
     pub discovery_details: String,
 }
 
+/// This defines a workload that should be scheduled to nodes
+/// that can access a capability described by this Configuration.
+/// The enum contents are boxed to use the heap instead of the stack.
+/// See this clippy warning for more details:
+/// https://rust-lang.github.io/rust-clippy/master/index.html#large_enum_variant
+#[derive(Serialize, Deserialize, PartialEq, Clone, Debug, JsonSchema)]
+#[serde(rename_all = "camelCase")]
+pub enum BrokerSpec {
+    // PodSpec for Pod that should be scheduled to all nodes that can
+    // access any capability described by this Configuration
+    BrokerPodSpec(Box<PodSpec>),
+    // JobSpec for Job that should be deployed to each capability described by this Configuration
+    BrokerJobSpec(Box<JobSpec>),
```

Comment:
nit: no need to prefix BrokerSpec objects with 'Broker' ... maybe `JobSpec` and `PodSpec` are enough?

Reply:
I added the prefix in case we run into issue with templating for OpenAPI or other specs.